velox/functions/lib/DateTimeFormatter.cpp

```diff
@@ -941,31 +1025,35 @@ std::string DateTimeFormatter::format(
   const date::year_month_day calDate(daysTimePoint);
   const date::weekday weekday(daysTimePoint);
 
-  std::string result;
+  char* resultBuffer = const_cast<char*>(result.data());
+  common::OutputByteStream outStream(resultBuffer);
+  std::string piece;
   for (auto& token : tokens_) {
     if (token.type == DateTimeToken::Type::kLiteral) {
-      result += token.literal;
+      outStream.append(token.literal.data(), token.literal.size());
     } else {
       switch (token.pattern.specifier) {
         case DateTimeFormatSpecifier::ERA:
-          result += static_cast<signed>(calDate.year()) > 0 ? "AD" : "BC";
+          piece = static_cast<signed>(calDate.year()) > 0 ? "AD" : "BC";
+          outStream.append(piece.data(), piece.length());
           break;
-
         case DateTimeFormatSpecifier::CENTURY_OF_ERA: {
           auto year = static_cast<signed>(calDate.year());
           year = (year < 0 ? -year : year);
           auto century = year / 100;
-          result += padContent(century, '0', token.pattern.minRepresentDigits);
+          piece = padContent(century, '0', token.pattern.minRepresentDigits);
```

Comment:
can we avoid creating intermediate strings?

Reply:
To also avoid intermediate strings inside `padContent()`, I just pass an `OutputByteStream` to `padContent` to directly write the content to result buffer in latest commit. Not sure whether it is a good implementation.